app/services/vine/api_service.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "faraday"
+
+module Vine
+  class ApiService
+    attr_reader :api_key, :jwt_generator
+
+    def initialize(api_key:, jwt_generator:)
```

Comment:
It's a way to isolate what `ApiService` depend on, see:  https://sandimetz.com/blog/2009/03/21/solid-design-principles/
This way, if we need to change the `jwt_generator` down the line, we can just inject a new one and we don't have to touch `ApiService`. It also makes testing easier because I can just pass a mock `jwt_generator`, I don't have to go and stub `generate_token`.

Reply:
Makes sense, thanks